test/fakeintake/client/client.go

```diff
@@ -121,17 +125,16 @@ func (c *Client) GetLatestFlare() (flare.Flare, error) {
 }
 
 func (c *Client) getFakePayloads(endpoint string) (rawPayloads []api.Payload, err error) {
-	resp, err := http.Get(fmt.Sprintf("%s/fakeintake/payloads?endpoint=%s", c.fakeIntakeURL, endpoint))
-	if err != nil {
-		return nil, err
-	}
-
-	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("error querying fake payloads, status code %s", resp.Status)
-	}
-
-	defer resp.Body.Close()
-	body, err := io.ReadAll(resp.Body)
+	var body []byte
+	err = backoff.Retry(func() error {
+		tmpResp, err := http.Get(fmt.Sprintf("%s/fakeintake/payloads?endpoint=%s", c.fakeIntakeURL, endpoint))
+		if err != nil || tmpResp.StatusCode != http.StatusOK {
+			return err
+		}
+		defer tmpResp.Body.Close()
```

Comment:
I think you can have non empty body for `StatusCode != http.StatusOK` so `defer tmpResp.Body.Close()` should be called after `if err != nil ` but before `tmpResp.StatusCode != http.StatusOK`